service/common/src/main/java/org/apache/polaris/service/task/BatchFileCleanupTaskHandler.java

```diff
@@ -77,17 +77,12 @@ public boolean handleTask(TaskEntity task, CallContext callContext) {
                 missingFiles.size());
       }
 
-      // Schedule the deletion for each file asynchronously
-      List<CompletableFuture<Void>> deleteFutures =
-          validFiles.stream()
-              .map(file -> super.tryDelete(tableId, authorizedFileIO, null, file, null, 1))
-              .toList();
+      CompletableFuture<Void> deleteFutures =
```

Comment:
nit: this is now a single future, so maybe we should name this `deleteFuture` instead